src/app/components/AccountPage/AccountPage.jsx

```diff
@@ -20,63 +23,33 @@ const AccountPage = (props) => {
     payload: newContent,
   });
 
-  // Build object in the format used by Legacy Catalog
-  // [input[name]]: input[value]
-  const [selectedItems, updateSelectedItems] = useState({});
-
-  const [errorMessage, updateErrorMessage] = useState(null);
+  const [errorMessage, setErrorMessage] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
+    if (errorMessage) console.error(errorMessage);
     if (!patron.id) {
       const fullUrl = encodeURIComponent(window.location.href);
       window.location.replace(`${appConfig.loginUrl}?redirect_uri=${fullUrl}`);
     }
     const accountPageContent = document.getElementById('account-page-content');
 
-    if (!accountPageContent) return;
-    const links = accountPageContent.getElementsByTagName('A');
-    const checkboxes = accountPageContent.querySelectorAll('input[type=checkbox]');
-
-    // this "Ratings" feature is in the html, but is not in use
-    accountPageContent.getElementsByTagName('th').forEach((th) => {
-      if (th.textContent.includes('Ratings')) th.style.display = 'none';
-    });
-    accountPageContent.getElementsByClassName('patFuncRating').forEach((el) => { el.style.display = 'none'; });
-
-    const errorMessageEls = accountPageContent.getElementsByClassName('errormessage');
-    if (errorMessageEls.length) {
-      // in original HTML this is `hidden`
-      errorMessageEls[0].style.display = 'block';
+    if (accountPageContent) {
+      manipulateAccountPage(
+        accountPageContent,
+        updateAccountHtml,
+        setErrorMessage,
+        patron,
+        content,
+        setIsLoading,
+      );
     }
-
-    checkboxes.forEach((checkbox) => {
-      checkbox.addEventListener('change', (e) => {
-        const { checked, name, value } = e.target;
-        updateSelectedItems((prevSelectedItems) => {
-          if (checked) {
-            prevSelectedItems[name] = value;
-            return prevSelectedItems;
-          }
-          delete prevSelectedItems[name];
-          return prevSelectedItems;
-        });
-      });
-    });
-
-    addEventListenersToAccountLinks(
-      links,
-      updateAccountHtml,
-      updateErrorMessage,
-      patron,
-      selectedItems,
-      content,
-    );
   });
```

Comment:
This `useEffect` is running multiple times so all the DOM elements are being manipulated multiple times - that's probably not what you want. If you want to run it only once, you can add `[]` as a second argument to the function. So something like

```
useEffect(() => { ....}, []);
```

The problem is that you're also checking for `patron.id`. That's not a variable that will change often, or at all, but if the intention is to keep that if statement there then you should add `patron` to the dependency array:

```
useEffect(() => { ....}, [patron]);
```

This might also be true for `errorMessage` but that seems like a temporary debugging statement.

If you expect users to leave this page and go to another page in the app, it'll be ideal if you also return a function that removes the event listeners. This is the cleanup function that is similar to `componentWillUnmount`.

```
useEffect(() => {
  ....
  return cleanupFunction();
}, [patron]);
```

We can go over this on a call. I'm thinking maybe splitting it up into two `useEffects` functions might be helpful but I need more context on what's going on.

Reply:
I ended up adding the `accountHtml` property to the second argument, since every time that changes `manipulateAccountPage` needs to be called. Thank you for checking this out!